addons/sales_team/models/crm_team.py

```diff
@@ -18,6 +18,7 @@ class CrmTeam(models.Model):
     _inherit = ['mail.thread']
     _description = "Sales Team"
     _order = "sequence"
+    _check_company_auto = True
```

Comment:
users are bound to switch company pretty much all the time, so I don't see a reason to do this. You might have users members of a sales team that switch between companies and this would fuck it up.

Reply:
never mind that comment, a3281058b080478451611b0c377fea243faf4cd8 makes it work correctly by checking the allowed companies instead of the current one of the user